components/calendar/src/any_calendar.rs

```diff
@@ -12,23 +12,13 @@ use crate::cal::{
 use crate::error::DateError;
 use crate::{types, AsCalendar, Calendar, Date, DateDuration, DateDurationUnit, Ref};
 
-use crate::preferences::{CalendarAlgorithm, HijriCalendarAlgorithm};
-use icu_locale_core::preferences::define_preferences;
-use icu_locale_core::subtags::language;
+use icu_locale_core::preferences::extensions::unicode::keywords::{
+    CalendarAlgorithm, HijriCalendarAlgorithm,
+};
 use icu_provider::prelude::*;
 
 use core::fmt;
 
-define_preferences!(
-    /// The prefs for date formatting.
-    [Copy]
-    AnyCalendarPreferences,
```

Comment:
Hm? Why rm AnyCalendarPreferences?

Reply:
DateTimeFormatter doesn't need the locale/preferences constructor anymore, because it validates its preferences and then constructs from `AnyCalendarKind`. I don't see a good use case for creating an `AnyCalendar` from a locale outside of the DTF path, so I've removed this for now. We can always add it back if it's needed.